src/does-match.ts

```diff
@@ -0,0 +1,29 @@
+import { deepEquals } from "./deep-equals"
+import { entries } from "./entries"
+import { includes } from "./includes"
+
+export const doesMatch = (target: object, value: object): boolean => {
+
+  const valueKeys = Object.keys(value)
+  const commonKeys = Object.keys(target).filter(value => includes(valueKeys, value))
+  const targetKV = entries(target)
+  const valueKV = entries(value)
+  const checkCommonKeys = (entree: any[][]):[string, unknown][] => {
```

Comment:
1/ Ok, never use any, never.
If you need to say "whatever type IDC" then use "unknown" please.
2/ use ReadonlyArray\<Whatever\> not Whatever[] please
3/ Extract all lambdas from here now please, performance :D

Reply:
Done